backend/command/src/items/events/item-edited.handler.ts

```diff
@@ -0,0 +1,132 @@
+import { InjectModel } from '@m8a/nestjs-typegoose';
+import { InternalServerErrorException, Logger } from '@nestjs/common';
+import { EventsHandler, IEventHandler } from '@nestjs/cqrs';
+import { ReturnModelType } from '@typegoose/typegoose';
+import { AnyBulkWriteOperation } from 'mongodb';
+import {
+  SuggestionItem,
+  SuggestionTag,
+} from '../../models/edit-suggestion.model';
+import { Item } from '../../models/item.model';
+import { Tag } from '../../models/tag.model';
+import { ItemEditedEvent } from './item-edited.event';
+
+@EventsHandler(ItemEditedEvent)
+export class ItemEditedHandler implements IEventHandler<ItemEditedEvent> {
+  private readonly logger = new Logger(ItemEditedHandler.name);
+  constructor(
+    @InjectModel(Item)
+    private readonly itemModel: ReturnModelType<typeof Item>,
+    @InjectModel(Tag)
+    private readonly tagModel: ReturnModelType<typeof Tag>,
+  ) {}
+  async handle({ itemEditedEventDto }: ItemEditedEvent) {
+    const updateItemStartTime = performance.now();
+
+    const { itemSlug, editValues } = itemEditedEventDto;
+
+    await this.updateItem(itemSlug, editValues);
+
+    this.logger.debug(
+      `${ItemEditedHandler.name} Item update took: ${
+        performance.now() - updateItemStartTime
+      }ms`,
+    );
+
+    // No tags in suggestion => we are done here
+    if (!editValues.tags) {
+      this.logger.debug(
+        `Total duration of ${ItemEditedHandler.name} was ${
+          performance.now() - updateItemStartTime
+        } ms`,
+      );
+      return;
+    }
+
+    const tags = editValues.tags;
+
+    const updateTagsStartTime = performance.now();
+    await this.updateTags(tags);
+    this.logger.debug(
+      `Updating tags by themselves took ${
+        performance.now() - updateTagsStartTime
+      } ms`,
+    );
+
+    this.logger.debug(
+      `Total duration of ${ItemEditedHandler.name} was ${
+        performance.now() - updateItemStartTime
+      } ms`,
+    );
+  }
+
+  // Db calls: Min: 1 findOneAndUpdate Max: 2 findOneAndUpdate()
+  async updateItem(slug: string, itemData: SuggestionItem) {
+    const { tags: tagsValues, weight, ...itemValues } = itemData;
+    const tagNamesToPull = tagsValues?.pull || [];
+    const tagsToPush =
+      tagsValues?.push?.map((tag) => ({ name: tag, count: 1 })) || [];
+    const weightSet = {};
+    if (weight !== undefined) {
+      // Generate object for nested update
+      // Format as described in: https://www.mongodb.com/docs/manual/reference/operator/update/set/#set-fields-in-embedded-documents
+      Object.keys(weight).forEach((key) => {
+        weightSet[`weight.${key}`] = weight[key];
+      });
```

Comment:
Technically it would be better to also use the name dynamically here to prevent a rename of the weight property to cause any errors.
remember, we hate Hardcoded strings lol.
Don't even know if this works, if  any solution would be longer than this pls ignore
```suggestion
      Object.keys(weight).forEach((key) => {
        weightSet[`${weight.name}.${key}`] = weight[key];
      });
```

Reply:
I don't think there is a nice and "pretty" way of doing this